api/v1beta1/machine_webhook.go

```diff
@@ -143,3 +146,17 @@ func (m *Machine) validate(old *Machine) error {
 	}
 	return apierrors.NewInvalid(GroupVersion.WithKind("Machine").GroupKind(), m.Name, allErrs)
 }
+
+func isMachinePoolMachine(m *Machine) bool {
+	if m.OwnerReferences == nil {
```

Comment:
You could omit this check since it's legal to `range` over a `nil` slice (iterating 0 times).